indico/modules/receipts/controllers/event.py

```diff
@@ -68,6 +75,65 @@ def _process(self):
         return jsonify(templates)
 
 
+@memoize_redis(3600)
+def _make_image_preview(img):
+    def _process_img(preview):
+        preview = square(preview)
+        if preview.height > IMAGE_PREVIEW_SIZE:
+            preview = preview.resize((IMAGE_PREVIEW_SIZE, IMAGE_PREVIEW_SIZE), resample=Image.BICUBIC)
+        # properly remove transparencies to prepare for JPEG conversion
+        if preview.mode in {'RGBA', 'LA'}:
+            background = Image.new(preview.mode[:-1], preview.size, 'white')
+            background.paste(preview, preview.split()[-1])
+            preview = background
+        image_bytes = BytesIO()
+        preview.save(image_bytes, 'JPEG')
+        image_bytes.seek(0)
+        return base64.b64encode(image_bytes.read())
+
+    if isinstance(img, bytes):
+        with Image.open(BytesIO(img)) as preview:
+            return _process_img(preview)
+    try:
+        with img.open() as f, Image.open(f) as preview:
+            if preview.format.lower() not in {'jpeg', 'png', 'gif', 'webp'}:
+                return ''
+            return _process_img(preview)
+    except OSError:
+        return ''
+
+
+class RHEventImages(RHManageRegFormsBase):
```

Comment:
Should this inherit from `RHManageRegFormsBase` or just `RHManageEventBase`?

Reply:
`RHManageRegFormsBase` has the advantage that it also grants access when you are *just* a registration manager without being a full event manager